core-java-modules/core-java-lang-oop-patterns/src/main/java/com/baeldung/cloning/Country.java

```diff
@@ -0,0 +1,72 @@
+package com.baeldung.cloning;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+public class Country implements Serializable {
+
+    private String code;
+    private String name;
+    private Currency currency;
+
+    public Country(String code, String name, Currency currency) {
+        this.code = code;
+        this.name = name;
+        this.currency = currency;
+    }
+
+    public Country(Country that) {
```

Comment:
maybe let's replace _that_ by _country_

_that_ can be a bit confusing

Reply:
check the latest commit. this is resolved